src/main/java/com/commercetools/sync/producttypes/utils/AttributeDefinitionsUpdateActionUtils.java

```diff
@@ -175,12 +174,13 @@
                               matchingNewAttributeDefinitionDraft.getAttributeType())) {
                             return buildActions(oldAttributeDefinition, attributeDefinitionDraft);
                           } else {
-                            // since there is no way to change an attribute type on CTP,
-                            // we remove the attribute definition and add a new one with a new
-                            // attribute type
-                            return Arrays.asList(
-                                RemoveAttributeDefinition.of(oldAttributeDefinitionName),
-                                AddAttributeDefinition.of(attributeDefinitionDraft));
+                            throw new UnsupportedOperationException(
```

Comment:
I would add here a comment which links to your ADR.

Reply:
https://github.com/commercetools/commercetools-sync-java/pull/787/commits/db7806ab1e770f0fe1bbaa9f6e5b40d311a9faf5